web3c/oasis.js

```diff
@@ -114,34 +110,27 @@ class Oasis {
     let accounts = web3.eth.accounts;
     let wrappedSigner = accounts.signTransaction.bind(accounts);
     let keyManager = this.keyManager;
-    accounts.signTransaction = function(tx, from) {
+    accounts.signTransaction = (tx, from) => {
       // Transaction to existing address, so we check it's confidential by asking the key manager.
       if (tx.to) {
-        return new Promise((resolve, reject) => {
-          keyManager.get(tx.to, function (err, publicKey) {
+        return new Promise(async (resolve, reject) => {
```

Comment:
an async function is an implicit promise, right?

Reply:
Yes. It wraps the return value in a promise. 

Like in the other comment below, we also want the `new Promise` here because we want to turn the `transformer.encryptTx`, which takes a callback, into a promise.  

In addition, we want the `async` so that we can `await` `this.isConfidential`.